### ISSUE TITLE
3D Distortion Problem

### ISSUE THREAD

**Issue by sant0ro** [sant0ro/habbo-camera-server#3]
The function rotate_polygon isn't done,. To do a 3D Polygon Distortion..

**Comment by sant0ro** [comment]
Observation: The problem is a 2D Distortion will only do a 0-360º degress rotation in CentreX and CentreY points. We need implement 3D distortion to Skrew the imagem (Flip the image in 3D to like that: http://www.imagemagick.org/Usage/img_diagrams/ewa_mapping.gif
Because the Wallpaper and FloorPaper Tiles must follow the CornerPoints and Wall and Floor direction.

**Comment by sant0ro** [comment]
What the real math Problem backend it?

1. PhP has really big glitches in his logical parameters, because that some mistakes happens in PhP.
2. In real math, an Image distortion by cartesian plane by X,Y points, is enough to redraw the corner of the polygon.
3. In "abstract" math, the realign must redraw the image, to filling the new polygon with the given angle.
4. But in PhP, the imagefilledpolygon doesn't understand that. Solution: add a 3D Coordinate (in this way "Z")
5. Is bad because, Pitagoras algorithm, and Gauss-Jordan Matricial algorithm must be enough, without using Z coordinate. 
6. Sadly, to redraw the fill (background) of polygon we need use something like A* algorithm, to trace the way of correct "background" direction.
7. Weirdly the Background fill exactly good the new Polygon. So is a PhP issue.
8. ImageRotate will not be good, because will rotate every tile in a 0º-360º degreess rotation, about himself centreX, centrY (cartesian) points. So that will not be enough.
9. ImageFlip only do VERTICAL and HORIZONTAL flips. So..
10. So, the idea is basically create a function to reassign the PolygonCorners Array, but that is already coded, but Php isn't getting to accert the right Background Position.
11. Real solution: Create algorithm based in X,Y position to find Z (really why?), but weirdly the jSON give to us a global Z position for that "tex-col", so: try to figure that out.

Anyone have a better idea?

**Comment by sant0ro** [comment]
Is a good idea using Kruskal?

**Comment by vista4life** [comment]
http://stackoverflow.com/questions/21213361/rotating-polygon-around-point

**Comment by sant0ro** [comment]
Yes, but that is not 3D distortion, is 2D polygon distortion around a single point, (this point is the center of the rotation)..

Soo that is not what i need.

**Comment by kylonx** [comment]
Claudio, why you do dis on me?!

**Comment by sant0ro** [comment]
sorry kylon i didn't understood what you're trying to say xD
